pkg/cmd/pipeline/translate.go

```diff
@@ -82,11 +82,13 @@ type CfgData struct {
 	Filename   string
 	Manifest   []byte
 	Icon       string
+	// ResourceName represents the pipeline name sanitized
+	ResourceName string
```

Comment:
Is there a way we don't include this variable in this struct? This struct is to keep all the information that will be stored *within* the configmap but this won't. 

Maybe we can omit this in the struct and let the handler to call the sanitization. At the end, this restriction is because we use Kubernetes as "storage" for the git deploy information. Sanitization call isn't too expensive, so I don't think there is a problem on calling it several times. The only drawback of this approach is to be aware of having to call it to build the config map name.

I would try to restrict this struct to the fields stored within the config map

Reply:
i've reverted the change and now the function to format the string is called whenever is needed